lib/mito/utilities/Shareable.h

```diff
@@ -42,20 +39,22 @@ namespace mito::utilities {
         inline Shareable & operator=(Shareable &&) = delete;
 
       private:
+        // accessor for the number of outstanding references
+        inline auto _references() const -> int;
         // increment the reference count
-        inline auto _acquire() -> int;
+        inline auto _acquire() const -> int;
         // decrement the reference count
-        inline auto _release() -> int;
+        inline auto _release() const -> int;
 
         // data members
       private:
-        int _reference_count;
+        mutable int _reference_count;
 
       private:
         // friendship with SharedPointer (the shared pointer needs r/w access to the reference count
         // of the Shareable instance)
         template <class T>
-        requires mito::utilities::ReferenceCountedObject<T>
+        // requires mito::utilities::ReferenceCountedObject<T>
```

Comment:
remove commented code if not needed

Reply:
The {requires} statement has been reinstated.